src/mcts/stoppers/legacy.cc

```diff
@@ -113,16 +113,15 @@ std::unique_ptr<SearchStopper> LegacyTimeManager::GetStopper(
   }
 
   // Total time, including increments, until time control.
-  auto total_moves_time =
-      std::max(0.0f, *time + increment * (movestogo - 1) - move_overhead_);
+  auto total_moves_time = std::max(0.0f, *time + increment * (movestogo - 1) -
+                                             move_overhead_ - time_spared_ms_);
 
   // If there is time spared from previous searches, the `time_to_squander` part
   // of it will be used immediately, remove that from planning.
   int time_to_squander = 0;
   if (time_spared_ms_ > 0) {
     time_to_squander = time_spared_ms_ * spend_saved_time_;
     time_spared_ms_ -= time_to_squander;
-    total_moves_time -= time_to_squander;
```

Comment:
Shouldn’t `time_to_squander` then be added to this_move_time?

Reply:
Yes, it was already there. With the default value of `immediate-use` (that is 1) the behavior of the code up to this point is the same.